src/dune_rules/super_context.ml

```diff
@@ -112,38 +113,46 @@ end = struct
       | _ -> None)
 
   let get_impl t dir =
-    (* We recompute the scope on every recursive call, even though it should be
-       unchanged. If this becomes a problem, we can memoize [find_by_dir]. *)
-    let scope = Scope.DB.find_by_dir t.scopes dir in
-    let inherit_from =
-      if Path.Build.equal dir (Scope.root scope) then
-        let format_config = Dune_project.format_config (Scope.project scope) in
-        Memo.lazy_ (fun () ->
-            let default_env = Memo.Lazy.force t.default_env in
-            Env_node.set_format_config default_env format_config)
-      else
-        match Path.Build.parent dir with
-        | None ->
-          Code_error.raise "Super_context.Env.get called on invalid directory"
-            [ ("dir", Path.Build.to_dyn dir) ]
-        | Some parent -> Memo.lazy_ (fun () -> get_node t ~dir:parent)
-    in
-    let config_stanza = get_env_stanza t ~dir in
-    let project = Scope.project scope in
-    let default_context_flags = default_context_flags t.context ~project in
-    let expander_for_artifacts =
-      Memo.lazy_ (fun () ->
-          expander_for_artifacts ~scope ~root_expander:t.root_expander
-            ~external_env:(external_env t ~dir) ~dir)
-    in
-    let expander =
-      Memo.Lazy.map expander_for_artifacts ~f:(fun expander_for_artifacts ->
-          extend_expander t ~dir ~expander_for_artifacts)
-    in
-    Env_node.make ~dir ~scope ~config_stanza ~inherit_from:(Some inherit_from)
-      ~profile:t.context.profile ~expander ~expander_for_artifacts
-      ~default_context_flags ~default_env:t.context_env
-      ~default_bin_artifacts:t.bin_artifacts
+    Memo.Build.return
```

Comment:
Can push down this call to `Env_node.make` below.

Reply:
Thanks, well-spotted!